src/Database/Transactions/TransactionManager.php

```diff
@@ -0,0 +1,16 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tempest\Database\Transactions;
+
+interface TransactionManager
+{
+    public function begin(): void;
+
+    public function commit(): void;
+
+    public function rollback(): void;
+
+    public function execute(callable $callback): bool;
```

Comment:
Am I correct in saying that `execute` is a shorthand for `begin`, `commit`, and `rollback`?

Maybe it would make more sense to have this method on the database interface instead of `Database::transaction()`. So you'd be able to do something like this

```php
$database->withinTransaction(function (TransactionManager $transaction) {

});
```

It would keep the `TransactionManager` interface a bit more cleaner, and it would make it so that you can immediately start a transaction from the database interface without another method in between.

Reply:
> Am I correct in saying that execute is a shorthand for begin, commit, and rollback?

Yuppers!

Hmmm, I kinda like this, but I'm not sure if I understand why this callback is getting an instance of `TransactionManager` - is `withinTransaction` supposed to be a replacement for `execute`, or just a callback that allows you to quickly get an instance of the `TransactionManager`? Meaning you'd still be responsible for calling commit?